pipeline/transformer/cifs.py

```diff
@@ -130,21 +133,68 @@ def _should_skip(self, situation, situationRecord):
 
         return False
 
-    def _detect_direction(self, situation, situationRecord):
+    def _laneStatusCoded(self, situationRecord: ET) -> Optional[str]:
+        """
+        Extracts the laneStatusCoded value of the given situationoRecord or None, if not available.
+
+        Example: for the following examplary, partial situationRecord, `o2xx` would be returned:
+
+        <situationRecord id="xxx">
+            <impact>
+                <impactExtension>
+                    <impactExtended>
+                        <laneStatusCoded>o2xx</laneStatusCoded>
+                        <laneRestriction>
+                            <lane>allLanesCompleteCarriageway</lane>
+                        </laneRestriction>
+                    </impactExtended>
+                </impactExtension>
+            </impact>
+        </situationRecord>
+        """
+        lsElement = situationRecord.find('d:impact/d:impactExtension/d:impactExtended/d:laneStatusCoded', ns)
+        return lsElement.text if lsElement is not None else None
+
+    @staticmethod
+    def _is_opposite_direction_concerned(lanestatus: str) -> bool:
+        # lanes can be single carriageways (encoded by centre line '1', or dual carriagways (encoded by '2'))
+        # we split at both.
+        lanesPerDirection = lanestatus.replace('2', '1').split('1')
+
+        leftLanes = lanesPerDirection[0]
+        rightLanes = lanesPerDirection[1]
+
+        # if leftLanes include more than unnrestricted lane, should, shoulder separatore, or
```

Comment:
I don't understand this comment. Can you try to clarify the cases the logic checks for, e.g. by splitting them into individual boolean variables which are then combined into a final check?

Reply:
Split into multiple boolean variables in [e4f36e5](https://github.com/mobidata-bw/ipl-dagster-pipeline/pull/108/commits/e4f36e5de5dc172138a2f7a7e4346dfdb54eede3)